storagemarket/testharness/dependencies/dependencies.go

```diff
@@ -141,6 +143,10 @@ func (gen *DepGenerator) New(
 	fs, err := filestore.NewLocalFileStore(filestore.OsPath(tempPath))
 	assert.NoError(t, err)
 
+	// We dont actually need to use the mount API in the mock DAG store
+	// for the provider
+	dagStore := dagstore.NewMockDagStoreWrapper(nil)
+
```

Comment:
Also, note that passing a nil  APU here is causing a nil exception in one of the tests:

```
aarshshah@Aarshs-MacBook-Pro go-fil-markets % go test -failfast ./... 
ok  	github.com/filecoin-project/go-fil-markets/carstore	0.591s
ok  	github.com/filecoin-project/go-fil-markets/dagstore	0.484s
?   	github.com/filecoin-project/go-fil-markets/dagstore/mocks	[no test files]
?   	github.com/filecoin-project/go-fil-markets/discovery	[no test files]
ok  	github.com/filecoin-project/go-fil-markets/discovery/impl	0.425s
?   	github.com/filecoin-project/go-fil-markets/discovery/migrations	[no test files]
?   	github.com/filecoin-project/go-fil-markets/docsgen	[no test files]
ok  	github.com/filecoin-project/go-fil-markets/filestore	0.228s
?   	github.com/filecoin-project/go-fil-markets/filestore/mocks	[no test files]
?   	github.com/filecoin-project/go-fil-markets/piecestore	[no test files]
ok  	github.com/filecoin-project/go-fil-markets/piecestore/impl	0.690s
?   	github.com/filecoin-project/go-fil-markets/piecestore/migrations	[no test files]
# github.com/filecoin-project/go-fil-markets/retrievalmarket.test
ld: warning: could not create compact unwind for _blst_sha256_block_data_order: does not use RBP or RSP based frame
# github.com/filecoin-project/go-fil-markets/storagemarket.test
ld: warning: could not create compact unwind for _blst_sha256_block_data_order: does not use RBP or RSP based frame
# github.com/filecoin-project/go-fil-markets/storagemarket/impl.test
ld: warning: could not create compact unwind for _blst_sha256_block_data_order: does not use RBP or RSP based frame

 Storage is complete
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x47a1dde]

goroutine 1377 [running]:
github.com/filecoin-project/go-fil-markets/dagstore.(*MockDagStoreWrapper).LoadShard(0xc0002a20f0, 0x5860790, 0xc000122010, 0xc0004842d0, 0x27, 0xc006e27df0, 0x5870e88, 0x560c580, 0x56395a4)
	/Users/aarshshah/go/src/github.com/filecoin-project/go-fil-markets/dagstore/mockdagstorewrapper.go:32 +0x3e
github.com/filecoin-project/go-fil-markets/retrievalmarket/impl.(*providerDealEnvironment).PrepareBlockstore(0xc000386690, 0x5860790, 0xc000122010, 0x1690b4da991e7b79, 0xc0004842d0, 0x27, 0x5586e40, 0x8)
	/Users/aarshshah/go/src/github.com/filecoin-project/go-fil-markets/retrievalmarket/impl/provider_environments.go:132 +0x86
github.com/filecoin-project/go-fil-markets/retrievalmarket/impl/providerstates.UnsealData(0xb8c0e98, 0xc0002b2f30, 0xb348c18, 0xc000386690, 0xc000484270, 0x22, 0x1690b4da991e7b79, 0xc000da20d8, 0x0, 0xc000468380, ...)
	/Users/aarshshah/go/src/github.com/filecoin-project/go-fil-markets/retrievalmarket/impl/providerstates/provider_states.go:32 +0x6f
reflect.Value.call(0x551fca0, 0x5797908, 0x13, 0x5637b81, 0x4, 0xc000607828, 0x3, 0x3, 0x400cd5f, 0x55c25e0, ...)
	/usr/local/Cellar/go/1.16.2/libexec/src/reflect/value.go:476 +0x8e7
reflect.Value.Call(0x551fca0, 0x5797908, 0x13, 0xc0003f4828, 0x3, 0x3, 0xc0003f4808, 0x427eeb0, 0x5572640)
	/usr/local/Cellar/go/1.16.2/libexec/src/reflect/value.go:337 +0xb9
github.com/filecoin-project/go-statemachine/fsm.fsmHandler.handler.func1(0xc000466990, 0x2, 0x2, 0xc000466990, 0xc0003f4a30, 0x407bb2f)
	/Users/aarshshah/go/pkg/mod/github.com/filecoin-project/go-statemachine@v0.0.0-20200925024713-05bd7c71fbfe/fsm/fsm.go:170 +0x437
reflect.Value.call(0xc0004342a0, 0xc000466930, 0x13, 0x5637b81, 0x4, 0xc0003f4f70, 0x2, 0x2, 0x400cd5f, 0x55748a0, ...)
	/usr/local/Cellar/go/1.16.2/libexec/src/reflect/value.go:476 +0x8e7
reflect.Value.Call(0xc0004342a0, 0xc000466930, 0x13, 0xc0002e0770, 0x2, 0x2, 0xc006e826e0, 0xc0002e0760, 0x449c928)
	/usr/local/Cellar/go/1.16.2/libexec/src/reflect/value.go:337 +0xb9
github.com/filecoin-project/go-statemachine.(*StateMachine).run.func3(0xc006e27d60, 0x5860790, 0xc000122010, 0xc006e27df0, 0xc006e27d70, 0xc0004341e0)
	/Users/aarshshah/go/pkg/mod/github.com/filecoin-project/go-statemachine@v0.0.0-20200925024713-05bd7c71fbfe/machine.go:102 +0x2c9
created by github.com/filecoin-project/go-statemachine.(*StateMachine).run
	/Users/aarshshah/go/pkg/mod/github.com/filecoin-project/go-statemachine@v0.0.0-20200925024713-05bd7c71fbfe/machine.go:100 +0x372
```

Reply:
It doesn't really make sense to use the real dag store here, because this is the storage market dependencies, but the dag store relies on both storage and retrieval deps (eg retrieval provider node).
Instead I've simplified the implementation of the dag store wrapper to just be stubbed out methods, and then override it in the retrieval market integration tests.
The panic is now fixed and the test is passing.